lib/Report/ReportTrait.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+namespace Xibo\Report;
+
+use Xibo\Factory\DisplayFactory;
+use Xibo\Factory\LayoutFactory;
+use Xibo\Factory\MediaFactory;
+use Xibo\Factory\SavedReportFactory;
+use Xibo\Service\ConfigServiceInterface;
+use Xibo\Service\DateServiceInterface;
+use Xibo\Service\LogServiceInterface;
+use Xibo\Service\SanitizerServiceInterface;
+use Xibo\Storage\StorageServiceInterface;
+use Xibo\Storage\TimeSeriesStoreInterface;
+use Slim\Http\Request;
+
+trait ReportTrait
+{
+
+    /**
+     * @var \Xibo\Helper\ApplicationState
+     */
+    private $state;
+
+    /**
+     * @var StorageServiceInterface
+     */
+    private $store;
+
+    /**
+     * @var TimeSeriesStoreInterface
+     */
+    private $timeSeriesStore;
+
+    /**
+     * @var LogServiceInterface
+     */
+    private $logService;
+
+    /**
+     * @var ConfigServiceInterface
+     */
+    private $configService;
+
+    /**
+     * @var DateServiceInterface
+     */
+    private $dateService;
+
+    /**
+     * @var SanitizerServiceInterface
+     */
+    private $sanitizerService;
+
+    /**
+     * @var DisplayFactory
+     */
+    private $displayFactory;
```

Comment:
I might be wrong, but it feels like these factories are specific to the report we are dealing with and not all reports will need them (and might need different ones).

This might be a good use case for a service locator factory where we have a `setDependencies($container)` method which takes the DI container as a parameter - each individual report class can then grab the factories it needs.

Reply:
This comment refers to the layout, media and display factories which would seem to be specific to the report we've made